pkg/generate/app/app.go

```diff
@@ -144,23 +144,28 @@ type BuildStrategyRef struct {
 }
 
 // BuildStrategy builds an OpenShift BuildStrategy from a BuildStrategyRef
-func (s *BuildStrategyRef) BuildStrategy() (*buildapi.BuildStrategy, []buildapi.BuildTriggerPolicy, error) {
-	triggers, err := s.Base.BuildTriggers()
+func (s *BuildStrategyRef) BuildStrategy() (*buildapi.BuildStrategy, []buildapi.BuildTriggerPolicy) {
 	if s.IsDockerBuild {
 		return &buildapi.BuildStrategy{
 			Type: buildapi.DockerBuildStrategyType,
-		}, triggers, err
+			DockerStrategy: &buildapi.DockerBuildStrategy{
+				From: &kapi.ObjectReference{
+					Kind: "ImageStreamTag",
+					Name: s.Base.String() + ":" + imageapi.DefaultImageTag,
+				},
+			},
+		}, s.Base.BuildTriggers()
 	}
 
 	return &buildapi.BuildStrategy{
 		Type: buildapi.STIBuildStrategyType,
 		STIStrategy: &buildapi.STIBuildStrategy{
 			From: &kapi.ObjectReference{
-				Kind: "DockerImage",
-				Name: s.Base.String(),
+				Kind: "ImageStreamTag",
```

Comment:
You also need to set the namespace if the image is from a different namespace.  So ImageRef needs to be able to tell you where it comes from - I would suggest having ImageRef generate an ObjectReference now (ObjectReference() *kapi.ObjectReference) and always return that, now that ben's changes are in.